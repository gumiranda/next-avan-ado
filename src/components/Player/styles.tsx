import styled from 'styled-components';
interface Props {
  empty?: boolean;
  isActive?: boolean;
}
export const BackgroundPlayer = styled.div`
  background: var(--purple-500);
`;
export const Wrapper = styled.div`
  padding: 4.5rem 5.5rem;
  width: 30rem;
  height: 82vh;
  background: var(--purple-500);
  color: var(--white);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
`;

export const HeaderPlayer = styled.header`
  display: flex;
  align-items: center;
  gap: 2rem;
`;
export const HeaderPlayerText = styled.strong`
  font-family: Lexend, sans-serif;
  font-weight: 600;
`;

export const EmptyPlayer = styled.div`
  height: 20rem;
  width: 100%;
  border: 1.5px dashed var(--purple-300);
  border-radius: 2.4rem;
  background: linear-gradient(
    143.8deg,
    rgba(145, 100, 250, 0.8) 0%,
    rgba(0, 0, 0, 0) 100%
  );
  padding: 6.4rem;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
`;
export const CurrentEpisode = styled.div`
  text-align: center;
  margin-top: 1rem;
  margin-bottom: 1rem;
  img {
    border-radius: 1.5rem;
  }
  strong {
    display: block;
    margin-top: 2rem;
    font: 600 1.25rem Lexend, sans-serif;
    line-height: 1.75rem;
  }
  span {
    display: block;
    margin-top: 1rem;
    opacity: 0.6;
    line-height: 1.5rem;
  }
`;

export const FooterPlayer = styled.footer`
  align-self: stretch;
  &.empty {
    opacity: ${(props: Props) => (props?.empty ? 0.5 : 1)};
  }
`;
export const FooterPlayerProgress = styled.div`
  display: flex;
  align-items: center;
  gap: 0.6rem;
  font-size: 0.875rem;
`;
export const FooterPlayerProgressTextIndicator = styled.span`
  display: inline-block;
  width: 4rem;
  text-align: center;
`;
export const FooterPlayerSlider = styled.div`
  width: 100%;
  height: 4px;
  background: var(--purple-300);
  border-radius: 4px;
  flex: 1;
`;
export const FooterPlayerEmptySlider = styled.div`
  width: 100%;
  height: 4px;
  background: var(--purple-300);
  border-radius: 4px;
`;
export const FooterPlayerButtons = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 3rem;
  gap: 3rem;
`;
export const GenericButton = styled.button`
  background: transparent;
  border: 0;
  font-size: 0;
  transition: filter 0.2s;
  filter: ${(props: Props) =>
    props.isActive
      ? 'invert(0.35) sepia(1) saturate(3) hue-rotate(100deg)'
      : 'brightness(0.95)'};
  &:disabled {
    cursor: default;
  }
  &:hover {
    filter: ${(props: Props) =>
      props.isActive
        ? 'brightness(0.55) invert(0.35) sepia(1) saturate(3) hue-rotate(100deg)'
        : 'brightness(0.95)'};
  }
  &:hover:not(:disabled) {
    filter: brightness(0.75);
  }
`;
export const PlayButton = styled.button`
  background: transparent;
  border: 0;
  font-size: 0;
  width: 4rem;
  height: 4rem;
  border-radius: 1rem;
  background: var(--purple-400);
  transition: filter 0.2s;
  &:hover:not(:disabled) {
    filter: brightness(0.75);
  }
  &:disabled {
    cursor: default;
  }
`;
