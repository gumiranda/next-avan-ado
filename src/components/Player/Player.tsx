import { PlayerContext } from 'domain/contexts';
import * as S from './styles';
import { useContext } from 'react';
const Player = () => {
  const { episodeList, currentEpisodeIndex } = useContext(PlayerContext);
  const episode = episodeList[currentEpisodeIndex] || {};
  return (
    <S.BackgroundPlayer>
      <S.Wrapper>
        <S.HeaderPlayer>
          <img src="/playing.svg" alt="Tocando agora" />
          <S.HeaderPlayerText>Tocando agora {episode?.title}</S.HeaderPlayerText>
        </S.HeaderPlayer>
        <S.EmptyPlayer>
          <strong>Selecione um podcast para ouvir</strong>
        </S.EmptyPlayer>
        <S.FooterPlayer>
          <S.FooterPlayerProgress>
            <S.FooterPlayerProgressTextIndicator>
              00:00
            </S.FooterPlayerProgressTextIndicator>
            <S.FooterPlayerSlider />
            <S.FooterPlayerEmptySlider />
            <S.FooterPlayerProgressTextIndicator>
              00:00
            </S.FooterPlayerProgressTextIndicator>
          </S.FooterPlayerProgress>
          <S.FooterPlayerButtons>
            <S.GenericButton type="button">
              <img src="/shuffle.svg" alt="Embaralhar" />
            </S.GenericButton>
            <S.GenericButton type="button">
              <img src="/play-previous.svg" alt="Tocar anterior" />
            </S.GenericButton>
            <S.PlayButton type="button">
              <img src="/play.svg" alt="Tocar" />
            </S.PlayButton>
            <S.GenericButton type="button">
              <img src="/play-next.svg" alt="Tocar prÃ³xima" />
            </S.GenericButton>
            <S.GenericButton type="button">
              <img src="/repeat.svg" alt="Repetir" />
            </S.GenericButton>
          </S.FooterPlayerButtons>
        </S.FooterPlayer>
      </S.Wrapper>
    </S.BackgroundPlayer>
  );
};
export default Player;
