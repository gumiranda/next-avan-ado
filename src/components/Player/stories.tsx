import { Story, Meta } from '@storybook/react/types-6-0'

import Player from './Player'

export default {
   title: 'Player',
   component: Player
} as Meta;

export const Basic: Story = (args) => <Player {...args} />;
