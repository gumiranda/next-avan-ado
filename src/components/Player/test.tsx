import { render, screen } from '@testing-library/react'

import Player from './Player'

describe('<Player />', () => {

  it('should render the heading', () => {
    const { container } = render(<Player />);
    expect(screen.getByRole('heading', { name: /Player/i })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  })

})
