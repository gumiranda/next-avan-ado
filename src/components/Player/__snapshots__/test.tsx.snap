// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> should render the logo image 1`] = `
.c0 {
  background: var(--purple-500);
}

.c1 {
  padding: 4.5rem 5.5rem;
  width: 30rem;
  height: 82vh;
  background: var(--purple-500);
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.2rem;
}

.c3 {
  font-family: Lexend,sans-serif;
  font-weight: 600;
}

.c4 {
  height: 20rem;
  width: 100%;
  border: 1.5px dashed var(--purple-300);
  border-radius: 2.4rem;
  background: linear-gradient( 143.8deg,rgba(145,100,250,0.8) 0%,rgba(0,0,0,0) 100% );
  padding: 6.4rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.c5.empty {
  opacity: 0.5;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.6rem;
  font-size: 0.875rem;
}

.c7 {
  display: inline-block;
  width: 4rem;
  text-align: center;
}

.c8 {
  width: 100%;
  height: 4px;
  background: var(--purple-300);
  border-radius: 4px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c9 {
  width: 100%;
  height: 4px;
  background: var(--purple-300);
  border-radius: 4px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 3rem;
  gap: 3rem;
}

.c11 {
  background: transparent;
  border: 0;
  font-size: 0;
}

.c12 {
  background: transparent;
  border: 0;
  font-size: 0;
  width: 4rem;
  height: 4rem;
  border-radius: 1rem;
  background: var(--purple-400);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <header
      class="c2"
    >
      <img
        alt="Tocando agora"
        src="/playing.svg"
      />
      <strong
        class="c3"
      >
        Tocando agora
      </strong>
    </header>
    <div
      class="c4"
    >
      <strong>
        Selecione um podcast para ouvir
      </strong>
    </div>
    <footer
      class="c5"
    >
      <div
        class="c6"
      >
        <span
          class="c7"
        >
          00:00
        </span>
        <div
          class="c8"
        />
        <div
          class="c9"
        />
        <span
          class="c7"
        >
          00:00
        </span>
      </div>
      <div
        class="c10"
      >
        <button
          class="c11"
          type="button"
        >
          <img
            alt="Embaralhar"
            src="/shuffle.svg"
          />
        </button>
        <button
          class="c11"
          type="button"
        >
          <img
            alt="Tocar anterior"
            src="/play-previous.svg"
          />
        </button>
        <button
          class="c12"
          type="button"
        >
          <img
            alt="Tocar"
            src="/play.svg"
          />
        </button>
        <button
          class="c11"
          type="button"
        >
          <img
            alt="Tocar prÃ³xima"
            src="/play-next.svg"
          />
        </button>
        <button
          class="c11"
          type="button"
        >
          <img
            alt="Repetir"
            src="/repeat.svg"
          />
        </button>
      </div>
    </footer>
  </div>
</div>
`;
